api/types/src/bytecode.rs

```diff
@@ -142,14 +147,27 @@ pub trait Bytecode {
                     .map(|f| self.new_move_struct_field(f))
                     .collect(),
             ),
-            StructFieldInformation::DeclaredVariants(..) => {
-                // TODO(#13806): implement for enums. Currently we pretend they don't have fields
-                (false, vec![])
+            StructFieldInformation::DeclaredVariants(variant_defs) => {
+                // Special case for enum representation of option module
```

Comment:
nit: maybe comment can specify that we want to keep it backwards compatible so it is not treated as enum and is backed by a vector